crates/ra_hir/src/generics.rs

```diff
@@ -25,6 +25,14 @@ pub struct GenericParam {
 pub struct GenericParams {
     pub(crate) parent_params: Option<Arc<GenericParams>>,
     pub(crate) params: Vec<GenericParam>,
+    pub(crate) where_clauses: Vec<WhereClause>,
+}
+
+/// A where clause, i.e. `where Type: Trait`.
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct WhereClause {
```

Comment:
Should we name it `WherePredicate`? I use WhereClose for the whole collection of predicates. Or is it me who is wrong? :)

Reply:
Chalk calls it `WhereClause`, which is why I went with it. rustc calls it `WherePredicate` in HIR though, and just `Predicate` in `ty`. So... I guess `WherePredicate` wins? ;)

Reply:
I renamed it.